src/app/components/courseInfoModal/CustomCourseModal.tsx

```diff
@@ -0,0 +1,261 @@
+  FormControl,
+  FormHelperText,
+  Input,
+  List,
+  ListItem,
+  Modal,
+  ModalClose,
+  Sheet,
+  Textarea,
+  Tooltip,
+  Typography,
+} from "@mui/joy";
+import { ChangeEvent, useState } from "react";
+
+const MAX_DESCRIPTION_LENGTH = 2000;
+
+export interface CustomCourseModalProps {
+  isOpen: boolean;
+  onClose: (c: StoredCourse) => void;
+  defaultCourse?: StoredCourse | undefined;
+}
+
+type QuartersOffered = {
+  Fall: boolean;
+  Winter: boolean;
+  Spring: boolean;
+  Summer: boolean;
+};
+
+const allQuartersOffered = {
+  Fall: true,
+  Winter: true,
+  Spring: true,
+  Summer: true,
+};
+
+export default function CustomCourseModal({
+  isOpen,
+  onClose,
+  defaultCourse,
+}: CustomCourseModalProps) {
+  const [course, setCustomCourse] = useState<StoredCourse>(
+    defaultCourse ? { ...defaultCourse } : customCourse(),
+  );
+
+  const [quarters, setQuarters] = useState<QuartersOffered>({
+    ...allQuartersOffered,
+  });
+
+  // Errors
+  const [invalidCreditsError, setInvalidCreditsError] = useState(false);
+  const [tooShortError, setTooShortError] = useState(false);
+
+  // Handlers
+  const handleTitleChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setTooShortError(false);
+    setCustomCourse((prev) => {
+      return {
+        ...prev,
+        title: e.target.value,
+      };
+    });
+  };
+
+  const handleDescriptionChange = (e: ChangeEvent<HTMLTextAreaElement>) => {
+    const newDescription = truncateTitle(
+      e.target.value,
+      MAX_DESCRIPTION_LENGTH,
+    );
+    setCustomCourse((prev) => {
+      return {
+        ...prev,
+        description: newDescription.slice(0, MAX_DESCRIPTION_LENGTH),
+      };
+    });
+  };
+
+  const handleCreditChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setInvalidCreditsError(false);
+    setCustomCourse((prev) => {
+      return {
+        ...prev,
+        credits: parseInt(e.target.value) ?? 0,
+      };
+    });
+  };
+
+  const handleQuarterChange = (
+    e: ChangeEvent<HTMLInputElement>,
+    item: keyof QuartersOffered,
+  ) => {
+    setQuarters((prev) => {
+      return {
+        ...prev,
+        [item]: e.target.checked,
+      };
+    });
+  };
+
+  const onSave = () => {
+    // Validation
+    if (course.title.length == 0) {
+      setTooShortError(true);
+      return;
+    }
+
+    if (course.credits < 1 || course.credits > 15) {
+      setInvalidCreditsError(true);
+      return;
+    }
+
+    // Get all the quarters
+    const quartersOffered = Object.keys(quarters).filter(
+      (q) => quarters[q as keyof QuartersOffered],
+    );
+    course.quartersOffered = quartersOffered;
+    onClose(course);
+
+    // Reset custom course and quarters
+    setCustomCourse(customCourse());
+    setQuarters({ ...allQuartersOffered });
+  };
+
+  return (
+    <Modal
+      onClose={onClose}
+      open={isOpen}
+      sx={{ display: "flex", justifyContent: "center", alignItems: "center" }}
+      onKeyDown={(e) => e.key === "Enter" && onSave()}
```

Comment:
This should be removed. Currently it is not possible to add newlines to the course description. On 'Enter' behavior works best to end editing of input elements. Having it exit the modal feels a bit weird